sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rule/BeamCalcRule.java

```diff
@@ -37,7 +40,20 @@ private BeamCalcRule() {
 
   @Override
   public boolean matches(RelOptRuleCall x) {
-    return true;
+    boolean hasRexOver = false;
```

Comment:
Add a comment to explain what this piece of code is doing. Basically it stops converting in the OVER clause case.